### ISSUE TITLE
firstValueFrom/lastValueFrom is missing AbortSignal support

### ISSUE THREAD

**Issue by ronag** [ReactiveX/rxjs#6442]
Promise based API's should generally also accept a `signal` property which is passed an [`AbortSignal`](https://nodejs.org/api/globals.html#globals_class_abortsignal) instance.

**Comment by ronag** [comment]
@benjamingr

**Comment by benjamingr** [comment]
@benlesh is someone I trust in both RxJS and AbortSignal and has provided API feedback before.

In my opinion @ronag is in the right here and the API _should_ take a signal.

**Comment by benjamingr** [comment]
> Why do you think that is the correct way? Why do you think promise API's should support the AbortSignal?

That's a good question.

This is related to discussions (e.g. https://github.com/ReactiveX/rxjs/issues/5545 ) in RxJS about building on and supporting modern cross-platform primitives.

It is the "official guidance" of Node.js and WHATWG that the way to handle cancellation in promise returning APIs is with AbortSignal. We have also added AbortSignal support to all of our own APIs in recent versions.

We don't own the way people write code. RxJS is welcome to do something different from Node.js here - and we are not _remotely_ forcing (or even strongly suggesting) adoption.

I think @ronag's ask is reasonable (support AbortSignal) in APIs that return promises (like .forEach and firstValueFrom).

**Comment by benlesh** [comment]
I agree. I would also like to add it to forEach, but that needs to be more carefully considered, I think.

**Comment by benlesh** [comment]
I'll have a go at this soon.

**Comment by deadbeef84** [comment]
@benlesh I dont see how `AbortSignal` having bad performance in some environments should block supporting signals in rxjs. This would be similar to not supporting promises because callbacks are faster, no? If performance is that big of an issue, mention it in the docs so users can opt for other alternatives where performance is an issue.
